src/NuGet.Indexing/GalleryServiceImpl.cs

```diff
@@ -1,59 +1,16 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
-using Lucene.Net.Search;
-using Microsoft.Owin;
+
 using System;
 using System.Collections.Generic;
-using System.Diagnostics;
+using Lucene.Net.Search;
+using Newtonsoft.Json;
 
 namespace NuGet.Indexing
 {
     public class GalleryServiceImpl
     {
-        public static string Query(IOwinContext context, NuGetSearcherManager searcherManager)
-        {
-            Trace.TraceInformation("Search: {0}", context.Request.QueryString);
-
-            string q = context.Request.Query["q"] ?? string.Empty;
-
-            string sortBy = context.Request.Query["sortBy"] ?? string.Empty;
-
-            bool includePrerelease;
-            if (!bool.TryParse(context.Request.Query["prerelease"], out includePrerelease))
-            {
-                includePrerelease = false;
-            }
-
-            bool countOnly;
-            if (!bool.TryParse(context.Request.Query["countOnly"], out countOnly))
-            {
-                countOnly = false;
-            }
-
-            string feed = context.Request.Query["feed"];
-
-            int skip;
-            if (!int.TryParse(context.Request.Query["skip"], out skip))
-            {
-                skip = 0;
-            }
-
-            int take;
-            if (!int.TryParse(context.Request.Query["take"], out take))
-            {
-                take = 20;
-            }
-            
-            bool ignoreFilter;
-            if (!bool.TryParse(context.Request.Query["ignoreFilter"], out ignoreFilter))
-            {
-                ignoreFilter = false;
-            }
-
-            return QuerySearch(searcherManager, q, countOnly, includePrerelease, sortBy, skip, take, feed, ignoreFilter);
-        }
-
-        public static string QuerySearch(NuGetSearcherManager searcherManager, string q, bool countOnly, bool includePrerelease, string sortBy, int skip, int take, string feed, bool ignoreFilter)
+        public static void Search(JsonWriter jsonWriter, NuGetSearcherManager searcherManager, string q, bool countOnly, bool includePrerelease, string sortBy, int skip, int take, string feed, bool ignoreFilter)
```

Comment:
Think we need to improve parameter validation and perhaps set sane defaults when no values provided? Can be trackd in a separate issue though.

e.g. if `take < 1` set it to 20, stuff like that :)